### PR TITLE
Extract rubocop config to  simplycop

### PR DESCRIPTION
**Pull request by Mangala-Melkunde**
This PR moves all rubocop config changes to simplycop. which provides standard shared rubocop configuration for Simply Business applications.

Trello:
https://trello.com/c/HkpYsXm8/1342-2-extract-rubocop-rules-from-chopin

FYI @simplybusiness/limited-liability

  - [ ] engineering review

### CODE REVIEW

File: .idea/simplycop.iml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (apeacock1991) [inline .idea/simplycop.iml]: This is an IDE-specific file, please remove it from the PR 😅

File: .simplycop_rails.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Rails:
+  Enabled: true
+
+## Rails Cops
+Rails/TimeZone:
+  Enabled: true
+  Severity: warning
+  Exclude:
+    - spec/features/step_definitions/valid_timezone_steps.rb
```
Comment (sebastian-nanek) [inline .simplycop_rails.yml]: this is app-specific

File: .simplycop_rspec.yml [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  Enabled: true
+
+RSpec/BeEql:
+  Description: Check for expectations where `be(...)` can replace `eql(...)`.
+  Enabled: true
+
+RSpec/BeforeAfterAll:
+  Description: Check that before/after(:all) isn't being used.
+  Enabled: false
+  Exclude:
```
Comment (sebastian-nanek) [inline .simplycop_rspec.yml]: this is app-specific

File: .simplycop_rspec.yml [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    BE: IS
+    have: has
+    HAVE: HAS
+  IgnoredWords: []
+
+RSpec/ExpectActual:
+  Description: Checks for `expect(...)` calls containing literal values.
+  Enabled: true
+  Exclude:
+    - spec/routing/**/*
```
Comment (sebastian-nanek) [inline .simplycop_rspec.yml]: this is app-specific

File: .simplycop_security.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+AllCops:
+  ExtraDetails: true
+
+Security/CSRFTokenValidation:
+  Enabled: true
+  Details: >-
+    It is important to have authenticity token validation enabled.
+    if you need to disable it please check with InfoSec department first.
+Security/RejectAllRequestsLocal:
```
Comment (sebastian-nanek) [inline .simplycop_security.yml]: add a new line above that one

File: .idea/simplycop.iml (hunk 1 above)
Comment (Mangala-Melkunde) [reply .idea/simplycop.iml]: @apeacock1991 yeah by mistake. will do that

**Comment by Mangala-Melkunde** [comment]
[(2) Extract rubocop rules from Chopin](https://trello.com/c/HkpYsXm8/1342-2-extract-rubocop-rules-from-chopin)

File: .idea/simplycop.iml (hunk 1 above)
Comment (Mangala-Melkunde) [reply .idea/simplycop.iml]: resolved

File: .simplycop_rails.yml (hunk 2 above)
Comment (Mangala-Melkunde) [reply .simplycop_rails.yml]: resolved

File: .simplycop_rspec.yml (hunk 3 above)
Comment (Mangala-Melkunde) [reply .simplycop_rspec.yml]: resolved

File: .simplycop_security.yml (hunk 5 above)
Comment (Mangala-Melkunde) [reply .simplycop_security.yml]: resolved

File: .simplycop_rspec.yml (hunk 4 above)
Comment (Mangala-Melkunde) [reply .simplycop_rspec.yml]: done

File: .custom_simplycop.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require:
+  - './lib/simplycop/custom_cops/timecop_without_block.rb'
+  - rubocop-rspec
+  - rubocop-rails
+AllCops:
```
Comment (sebastian-nanek) [inline .custom_simplycop.yml]: empty line above

**Comment by sebastian-nanek** [comment]
Please move specs for custom cops from Chopin here.

File: .simplycop_rspec.yml [hunk 7]
```diff
@@ -0,0 +1,223 @@
+
```
Comment (sebastian-nanek) [inline .simplycop_rspec.yml]: empty line

**Comment by sebastian-nanek** [comment]
Separate card, I guess? @apeacock1991

**Comment by apeacock1991** [comment]
@sebastian-nanek I don't mind, it's a copy-and-paste job, and really we shouldn't merge the Chopin PR until SimplyCop is on Gemstash, as all our private gems should be sourced from there, so I would argue to JDI

**Pull request merged**